<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>yasav</class>
 <widget class="QWidget" name="yasav">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>yasav</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <widget class="arrayviewer" name="viewer" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>4</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>400</width>
       <height>300</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayoutSetup">
     <item>
      <widget class="QGroupBox" name="groupBoxControl">
       <property name="title">
        <string>Control</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutControls">
          <item>
           <widget class="QPushButton" name="pushButtonGoStop">
            <property name="text">
             <string>Go</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonIteration">
            <property name="text">
             <string>Iteration</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonStep">
            <property name="text">
             <string>Step</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutTiming">
          <item>
           <widget class="QLabel" name="labelTiming">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Step every</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBoxTiming">
            <property name="suffix">
             <string>ms</string>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
            <property name="value">
             <number>200</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxArraySettings">
       <property name="title">
        <string>Array</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QComboBox" name="comboBoxArrayInit"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QSlider" name="horizontalSliderArraySize">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>0</height>
             </size>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>200</number>
            </property>
            <property name="pageStep">
             <number>100</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBothSides</enum>
            </property>
            <property name="tickInterval">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelArraySize">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Array Size: (def)</string>
            </property>
            <property name="textFormat">
             <enum>Qt::PlainText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonShuffle">
          <property name="text">
           <string>Shuffle</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxSortingAlgorithm">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>3</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="title">
        <string>Sorting Algorithm</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QComboBox" name="comboBoxSortingAlgorithm"/>
        </item>
        <item>
         <widget class="QFrame" name="frameSortingAlgorithmDesc">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="labelA">
             <property name="text">
              <string notr="true">A = -1</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="labelB">
             <property name="text">
              <string notr="true">B = -1</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelArrA">
             <property name="text">
              <string notr="true">ARR[A] = -1</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="labelArrB">
             <property name="text">
              <string notr="true">ARR[B] = -1</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QLabel" name="labelNextStep">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
               <horstretch>0</horstretch>
               <verstretch>1</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>180</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>arrayviewer</class>
   <extends>QWidget</extends>
   <header>array/arrayviewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>horizontalSliderArraySize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>yasav</receiver>
   <slot>updateArraySizeLabel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>758</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>699</x>
     <y>545</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonShuffle</sender>
   <signal>released()</signal>
   <receiver>yasav</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>777</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>899</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderArraySize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>yasav</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>777</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>896</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxArrayInit</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>yasav</receiver>
   <slot>updateArrayFill(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>777</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>900</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonGoStop</sender>
   <signal>clicked()</signal>
   <receiver>yasav</receiver>
   <slot>toggleGoStop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>604</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>898</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStep</sender>
   <signal>released()</signal>
   <receiver>yasav</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>776</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxTiming</sender>
   <signal>valueChanged(int)</signal>
   <receiver>yasav</receiver>
   <slot>updateTicker(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>776</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>896</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxSortingAlgorithm</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>yasav</receiver>
   <slot>updateSorter(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>777</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>897</x>
     <y>357</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxSortingAlgorithm</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>yasav</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>777</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>897</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonIteration</sender>
   <signal>released()</signal>
   <receiver>yasav</receiver>
   <slot>toggleIteration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>646</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>793</x>
     <y>7</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateArraySizeLabel(int)</slot>
  <slot>stop()</slot>
  <slot>updateArrayFill(int)</slot>
  <slot>toggleGoStop()</slot>
  <slot>updateTicker(int)</slot>
  <slot>updateSorter(int)</slot>
  <slot>toggleIteration()</slot>
 </slots>
</ui>
